### PR TITLE
Refactoring with solid

### PR DESCRIPTION
**Pull request by huseyinbiyik**
You can find the features added for this Pull Request below.
Built by @huseyinbiyik  @Arturgouveia1970 

- YAGNI, KISS and DRY rules and Ruby style guide are respected.
- Followed SOLID principles.
- Paid special attention to the "Single-responsibility Principle".
- Make sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).
- Extracted some pieces of logic into a separate class(es).

Thank you for the review in advance! 🤓

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -14,6 +15,12 @@ def initialize
...
 
+  def greet
+    puts 'Welcome to the School Library App!'
+    puts "\n"
+    menu
+  end
+
   # list all books
   def list_books
     if @books.empty?
```
Comment (omarsalem7) [inline app.rb:28]: - [x] Kindly separate functions in `app.rb` in several classes as the Single-Responsibility Principle of the SOLID object Oriented Design states that a class should have one job only. hence all requests for inputs(`create_person`,`create_student`,`create_teacher`,`create_book`,`create_rental`) should have been placed in the separate files whose responsibilities are to request input. Please kindly work on this and move a step forward toward approval.

For more on the SOLID lessons, click on this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid)

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @huseyinbiyik, @Arturgouveia1970 😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights
- Descriptive pull request ✔️
- No linter errors ✔️
- Good descriptive readme ✔️

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (huseyinbiyik) [reply app.rb:28]: @omarsalem7  Thank you for the review. Fixed it now.

**Review by dasileker — APPROVED** [review]
### **Hi @Arturgouveia1970,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**